coast/TIDEGAUGE.py

```diff
@@ -0,0 +1,395 @@
+                        qc_flags.append(int(working_line[3]))
+                    
+                line_count = line_count + 1
+             
+        # Convert time list to datetimes using pandas
+        time = np.array(pd.to_datetime(time))
+        
+        # Return only values between stated dates
+        start_index = 0
+        end_index = len(time)
+        if date_start is not None:
+            date_start = np.datetime64(date_start)
+            start_index = np.argmax(time>=date_start)
+        if date_end is not None:
+            date_end = np.datetime64(date_end)
+            end_index = np.argmax(time>date_end)
+        time = time[start_index:end_index]
+        sea_level = sea_level[start_index:end_index]
+        qc_flags=qc_flags[start_index:end_index]
+        
+        # Set null values to nan
+        sea_level = np.array(sea_level)
+        qc_flags = np.array(qc_flags)
+        sea_level[qc_flags==5] = np.nan
+        
+        # Assign arrays to Dataset
+        dataset['time'] = xr.DataArray(time, dims=['time'])
+        dataset['sea_level'] = xr.DataArray(sea_level, dims=['time'])
+        dataset['qc_flags'] = xr.DataArray(qc_flags, dims=['time'])
+        
+        # Assign local dataset to object-scope dataset
+        return dataset
+    
+    def plot_map(self):
+        '''
+        Plot tide gauge locations on a map
+        
+        Example usage:
+        --------------
+        # For a TIDEGAUGE object tg
+        tg.plot_map()
+
+        '''
+        try:
+            import cartopy.crs as ccrs  # mapping plots
+            import cartopy.feature  # add rivers, regional boundaries etc
+            from cartopy.mpl.gridliner import LONGITUDE_FORMATTER, LATITUDE_FORMATTER  # deg symb
+            from cartopy.feature import NaturalEarthFeature  # fine resolution coastline
+        except ImportError:
+            import sys
+            warn("No cartopy found - please run\nconda install -c conda-forge cartopy")
+            sys.exit(-1)
+
+        import matplotlib.pyplot as plt
+        fig = plt.figure(figsize=(10, 10))
+        ax = fig.gca()
+        ax = plt.subplot(1, 1, 1, projection=ccrs.PlateCarree())
+
+        cset = plt.scatter(self.longitude, self.latitude, c='k')
+
+        ax.add_feature(cartopy.feature.BORDERS, linestyle=':')
+        coast = NaturalEarthFeature(category='physical', scale='50m',
+                                    facecolor=[0.8,0.8,0.8], name='coastline',
+                                    alpha=0.5)
+        ax.add_feature(coast, edgecolor='gray')
+
+        gl = ax.gridlines(crs=ccrs.PlateCarree(), draw_labels=True,
+                          linewidth=0.5, color='gray', alpha=0.5, linestyle='-')
+
+        gl.top_labels = False
+        gl.bottom_labels = True
+        gl.right_labels = False
+        gl.left_labels = True
+        gl.xformatter = LONGITUDE_FORMATTER
+        gl.yformatter = LATITUDE_FORMATTER
+
```

Comment:
Add:
        plt.title('Map of gauge locations')
        plt.ylabel('Latitude') 
        plt.xlabel('Longitude')

Reply:
done